nx/lib/nx/shared.ex

```diff
@@ -225,6 +242,55 @@ defmodule Nx.Shared do
     end
   end
 
+  @doc """
+  BF16 write callback.
+  """
+  def write_f8(data) do
+    case data do
+      :infinity -> unquote(Nx.Type.infinity_binary({:f, 8}))
+      :neg_infinity -> unquote(Nx.Type.neg_infinity_binary({:f, 8}))
+      :nan -> unquote(Nx.Type.nan_binary({:f, 8}))
+      value -> Nx.Shared.erlang_float_to_fp8(value)
+    end
+  end
+
+  @doc """
+  Approximates an Erlang float as an 8-bit binary according to
+  the FP8 spec.
+  """
+  def erlang_float_to_fp8(float) do
+    {sign, pos_float} =
+      if float < 0,
+        do: {1, -float},
+        else: {0, float}
+
+    exponent = :math.log2(pos_float) |> Float.floor() |> Kernel.+(7) |> trunc()
+    mantissa = (Float.floor(pos_float / :math.pow(2, exponent - 7) * 8) - 8) |> trunc()
+
+    exponent_bin = exponent &&& 15
+    mantissa_bin = mantissa &&& 7
+
+    <<sign::size(1), exponent_bin::size(4), mantissa_bin::size(3)>>
+  end
+
+  @doc """
+  Converts an 8-bit binary back to an Erlang float.
+
+  We do this by expanding the 8-bit binary to a 16-bit binary
+  and then interpreting the binary as an F16 as that's the lowest
+  precision float supported by Erlang.
+  """
+  def fp8_to_erlang_float(<<sign::size(1), exponent_fp8::size(4), mantissa_fp8::size(3)>>) do
+    exponent_fp16 =
+      if exponent_fp8 == 0,
+        do: 0,
+        else: exponent_fp8 - 7 + 15
+
+    mantissa_fp16 = mantissa_fp8 <<< 7
+    <<x::float-16-big>> = <<sign::size(1), exponent_fp16::size(5), mantissa_fp16::size(10)>>
```

Comment:
We don't need to convert to a binary, do we? We could probably write the formula directly:

  V=(−1)^S×2^(E−bias)×(1+M/8​)

Or something like that?